src/common/common.cpp

```diff
@@ -234,4 +245,46 @@ bool drop_elevated_privileges() {
   CloseHandle(token);
 
   return result;
-}
\ No newline at end of file
+}
+
+std::wstring get_process_path(DWORD pid) noexcept {
+  auto process = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, TRUE, pid);
+  std::wstring name;
+  if (process != INVALID_HANDLE_VALUE) {
+    name.resize(MAX_PATH);
+    DWORD name_length = static_cast<DWORD>(name.length());
+    QueryFullProcessImageNameW(process, 0, (LPWSTR)name.data(), &name_length);
```

Comment:
Better to check the return value of QueryFullProcessImageNameW

Reply:
updated!